src/components/fetcher/index.js

```diff
@@ -40,11 +42,23 @@ const Fetcher = ({
         setFetching(false);
         return setNotification('streamReadFailure', deviceId);
       }
-      let sensor_data_packets = await getSensorData(mamStreamData);
+      const sensorData = await getSensorData(mamStreamData);
+      // Remove pruned (undefined) data packages
+      const filteredSensorData = sensorData.filter((data) => {
```

Comment:
For simplicity you could also write it like this: `const filteredSensorData = sensorData.filter(data => data);`